TicTacToeMain.cpp

```diff
@@ -97,6 +93,42 @@ int main(int argc,char* argv[])
     Log.LogMessage("Game Mode: " + to_string(mapGameSettings.at(objCommons->GameSettings_GameMode)));
     Log.LogMessage("Comuter Match Speed: " + to_string(mapGameSettings.at(objCommons->GameSettings_ComputerMatchSpeed)));
 
+    while(!bPlayableSelected)
+    {
+        cout << "Do you want to be X's or O's?" << endl;
+        cin >> sPlayable;
+
+        try
+        {
+            transform(sPlayable.begin(), sPlayable.end(), sPlayable.begin(), ::toupper);
+            if(sPlayable == "X")
+            {
+                mapGameSettings.insert(pair<Commons::GameSettings,Commons::Playable>(objCommons->GameSettings_PlayerType, objCommons->Playable_X));
+                //ObjPlayer_1->SetPlayable(objCommons->Playable_X);
+                //ObjPlayer_2->SetPlayable(objCommons->Playable_O);
+                bPlayableSelected = true;
+            }
+            else if(sPlayable == "O")
+            {
+                mapGameSettings.insert(pair<Commons::GameSettings,Commons::Playable>(objCommons->GameSettings_PlayerType, objCommons->Playable_O));
+                //ObjPlayer_1->SetPlayable(objCommons->Playable_O);
+                //ObjPlayer_2->SetPlayable(objCommons->Playable_X);
+                bPlayableSelected = true;
+            }
+            else
+            {
+                cout << "Bruh, pick X or O..." << endl;
+            }
+        }
+        catch(const std::exception& e)
```

Comment:
I don't think it's best practice to catch the base exception class.
If you are expecting a particular sub type of exception catch that else in the state of the game in one place you can catch all the exceptions.
Just something thing to think about, is not a necessary change.

Reply:
I think that is fair, I should make `TicTacToeMain.cpp` catch all the exceptions and everything else throw them. Since this is a bigger job, I'll create another issue on GitHub for the changes: https://github.com/nofea/TicTacToe/issues/61